### PR TITLE
Fix/digital copy access

### PR DESCRIPTION
**Pull request by Adamik10**
#### Link to issue
https://reload.atlassian.net/browse/DDFSOEG-303

#### Description
Similarly to #441 this PR fixes an access issue for users that aren't living in the same municipality as the library they are currently browsing. In this case it is in regards to the digital articles.

#### Screenshot of the result
n/a

#### Checklist

- [ ] My complies with [our coding guidelines](../docs/code_guidelines.md).
- [ ] My code passes our static analysis suite. If not then I have added a comment explaining why this change should be exempt from the code standards and process.
- [ ] My code passes our continuous integration process. If not then I have added a comment explaining why this change should be exempt from the code standards and process.

#### Additional comments or questions
n/a

### CODE REVIEW

File: src/components/material/material-buttons/online/MaterialButtonOnlineDigitalArticle.tsx [hunk 1]
```diff
@@ -17,6 +20,15 @@ export interface MaterialButtonOnlineDigitalArticleProps {
...
 > = ({ pid, size, dataCy = "material-button-online-digital-article" }) => {
+  const [isResident, setIsResident] = useState<null | boolean>(null);
+  const { isLoading } = useGetPatronInformationByPatronIdV2({
+    enabled: !isAnonymous(),
+    onSuccess: (data) => {
+      if (data?.patron?.resident !== undefined) {
+        setIsResident(data.patron.resident);
+      }
+    }
+  });
```
Comment (spaceo) [inline src/components/material/material-buttons/online/MaterialButtonOnlineDigitalArticle.tsx:31]: Didn't you talk with Kasper about creating a hook for fetching Patron data if user is logged in? Would make sense to use it here as well.

I think you should also create a function alongside the other user functions called ìsResident()` that looks at the `resident` property of the `Patron` object. We probably need it in other places in the future.

**Review by spaceo — CHANGES_REQUESTED** [review]
Great implementation! I have some remarks...

**Review by spaceo — COMMENTED** [review]
Great! One comment...

**Comment by spaceo** [comment]
Also there is a dear old Cypress test that is failing...

File: src/components/material/material-buttons/online/MaterialButtonOnlineInfomediaArticle.tsx [hunk 2]
```diff
@@ -27,18 +28,18 @@ const MaterialButtonOnlineInfomediaArticle: FC<
...
   const { openGuarded } = useModalButtonHandler();
   const { authUrl } = useUrls();
 
+  useDeepCompareEffect(() => {
+    if (!userData || !userData.patron) {
+      return;
+    }
+    setIsUserResident(isResident(userData?.patron));
+  }, [userData]);
+
```
[the remark this answers is not in the document]
Comment (Adamik10) [reply src/components/material/material-buttons/online/MaterialButtonOnlineInfomediaArticle.tsx:42]: Why is that? usePatronData() hook is asynchronous after all 🤔

**Review by spaceo — APPROVED** [review]
LGTM 👍

**Pull request merged**